### PR TITLE
Prelim Unit Tests

### LINKED ISSUE #16: Use JS to create overlay cookies

### PR DESCRIPTION
**Pull request by gfargo**
This PR creates the preliminary tests & changes needed to [prep `FM-Overlays` plugin for production](https://alleyinteractive.atlassian.net/browse/CIR-218).  The goal was to accomplish the following:
- Complete a basic Unit Test suite that covers the basic use cases
- Set cookies via Javascript - resolves #16 
- Update comments, Docblocks and remove any deprecated `@Todos` 
### Unit Tests

All unit tests should extend the base testing class `FM_Overlays_UnitTest`.  The current test suite checks the following:

**Basic**
Runs the follow basic tests ensuring this are being created & saving post meta correctly
- Basic Overlay Post Creation
- Cookie Name is being inserted into DOM correctly
- RichText Overlay post creation
- Saving RichText content to overlay post
- Image Overlay post creation 
- Image Type in DOM

**Conditional**
Tests for the following conditional classes 
- is_front_page
- is_home
- is_page
- is_tag
- has_tag
- is_cat
- has_cat
- is_single

**Priority**
The initial priority testing has been completed in this PR.  The current test will create two overlays, one with a higher `menu_order` and verify that it displays within the footer.

Related to: https://alleyinteractive.atlassian.net/browse/CIR-218

### CODE REVIEW

**Comment by gfargo** [comment]
This PR should effectively resolve #15, with 12 tests and 32 assertions.

File: client/js/global.js [hunk 1]
```diff
@@ -41,6 +42,20 @@ function fmOverlay() {
   }
 
   /**
+   * Set Overlay cookies.
+   */
+  function setCookie() {
+    const name = cookieName;
```
Comment (kjbenk) [inline client/js/global.js]: Why can't you just remove this variable and only use `cookieName`?

File: client/js/global.js [hunk 2]
```diff
@@ -41,6 +42,20 @@ function fmOverlay() {
...
+    const date = new Date();
+
+    // set cookie for 2 hours
+    date.setTime(date.getTime() + (2 * 60 * 60 * 1000));
+
+    const expires = `; expires=' + ${date.toGMTString()}`;
+    document.cookie = `${name} = ${true} ${expires}; path=/`;
```
Comment (kjbenk) [inline client/js/global.js]: Is `${true}` valid?  I thought that had to be a variable.

File: templates/fm-overlay-basic.php [hunk 3]
```diff
@@ -13,10 +13,14 @@
...
 $fm_overlay_classes = Fm_Overlays_Helpers::instance()->get_overlay_classes( $overlay );
+$targeted_conditions = Fm_Overlays::instance()->targeted_conditions;
 ?>
 
-<!-- @TODO: we should have a dynamic class and/or ID generated for styling purposes -->
-<div id="fm-overlay" class="<?php echo esc_attr( $fm_overlay_classes ) ?>">
+<div
+	id="fm-overlay"
+	class="<?php echo esc_attr( $fm_overlay_classes ) ?>"
+	data-cookiename="<?php echo sanitize_html_class( Fm_Overlays::instance()->get_overlay_cookie_name( $overlay->ID ) ); ?>"
```
Comment (kjbenk) [inline templates/fm-overlay-basic.php]: Why not use `esc_attr()` here since `data-cookiename` is not a list of class names...right?

**Comment by kjbenk** [comment]
Very good work, I had a couple minor questions, but otherwise :golf:

File: templates/fm-overlay-basic.php (hunk 3 above)
Comment (gfargo) [inline templates/fm-overlay-basic.php]: Ah great point, I'll swap it over to attr.

File: client/js/global.js (hunk 1 above)
Comment (gfargo) [inline client/js/global.js]: Totally can & will do; forgot to go back and refactor this after moving the scope of `cookieName` to the parent function. Thanks for catching this

**Comment by gfargo** [comment]
Thanks Kyle! Great suggestions, going to push up the changes you suggested and merge this bad boy into `master`.

File: client/js/global.js (hunk 2 above)
Comment (gfargo) [inline client/js/global.js]: AFAIK, we just want to set the cookies value to true - so this from my understanding will simply evaluate `true` to `true`.  

Could probably use static text here but hey, this way later one if we want to actually use the cookie to store values we can just swap `true` for a variable.

**Pull request merged**